feg/gateway/gtp/gtp_client.go

```diff
@@ -122,9 +125,34 @@ func (c *Client) Run(ctx context.Context) error {
 			return
 		}
 	}()
+	glog.V(2).Infof("Wait until local gtp client is ready")
+	waitUntilClientIsReady(c, 0)
 	return nil
 }
 
+// waitUntilClientIsReady is a hack to know when the client is ready and avoid null pointer issues using
+// the GTP-C client too early. Since go-gtp doesn't offer any visibuility on the readines of the connection
+// we use LocalAddrs as indicator
```

Comment:
Could you explain the logic here? Is the expected sequence: NewClient(), Enable(), Run()?
What exactly are you waiting for? What initializes gtp's c.pktConn?

Reply:
`NewCLient` just creates our magma type Client that wraps the go-gtp client. It does not initialize it. The reason is initialization is done in the other `New` functions which carry some helper methods to figure out ips.

`Enable`: creates the con Connection object.

We could have had `Enable` as part of `NewClient`. I just wanted to separate its logic 


The order is 
- any of the `New....` functions
- `Enable`
- `run`

This can be seen in `NewRunningClient`

Alternatively we have `NewConnectedClient` that doesn't use `Enable` and `run`, but it uses `Dia`.   
The difference between them is:
- `NewRunningClient`: it only starts the client. 
- `NewConnectedClient`: it checks if a specific pgw is available (using echo) and then starts the client



About the catching panic, the problem is when we start the client, the listener takes few milliseconds to be available. But that listener is hidden from us, so we can;t check when its ready. The only way to check it is with that panic.   

I will try to do a PR against Go-GTP and see if i can include a function on the `gtp.Client` that allows us to see if the initialization has been completed